scss/_button-group.scss

```diff
@@ -74,60 +78,37 @@
 .dropdown-toggle-split {
   padding-right: subtract($dropdown-padding-x, $border-width);
   padding-left: subtract($dropdown-padding-x, $border-width);
-  border-color: $black;
+  border-color: currentColor;
 
   // Boosted mod
-  &:not(:hover):not(:focus):not(:active) {
-    border-left-color: transparent;
-
-    &.btn-inverse {
-      color: $white;
-      background-color: $black;
-      border-color: $white;
+  .btn-group:not(.dropstart) &,
+  .dropstart & ~ .btn {
+    &:not(:hover):not(:focus):not(:active) {
+      border-left-color: transparent;
 
       &::before {
-        background-color: $black;
+        position: absolute;
+        top: -$border-width;
+        bottom: -$border-width;
+        left: -$border-width;
+        width: $border-width;
+        color: inherit;
+        content: "";
+        background:
+          linear-gradient(
+            currentColor $border-width,
+            transparent $border-width,
+            transparent $border-width * 2.5,
+            currentColor $border-width * 2.5,
+            currentColor subtract(100%, $border-width * 2.5),
+            transparent subtract(100%, $border-width * 2.5),
```

Comment:
We may need to create a CCS var for this "2.5" value if we use it multiple times.